### ISSUE TITLE
Updating files in media library gives new urls 

### ISSUE THREAD

**Issue by Aaen** [umbraco/Umbraco-CMS#9481]
A brief description of the issue goes here.

The issue here is when you e.g have a link to a pdf in the media library and want to update this one to a new version your new pdf file will have a new link and you need to make a redirect from the old URL to the new URL 

## Umbraco version

Umbraco 8.9.1 

### Steps to reproduce
1. Upload a file in the media library
2. Remove the file from the node (Don´t delete the node in the media library)
3. Chose a new file
4. Now you will see that the URL is another one than the old one.

### Expected result
I would expect that the URL will stay the same 

### Actual result
Now you will see that the URL is another one than the old one.

**Comment by ktzanoulinos** [comment]
on Umbraco version 8.5.5 we are experiencing the same issue.
In the past and in the same version, the URL did NOT change. Suddenly and without any change code wise or version wise, the URL changes when we remove the file and upload a new one (with the exact same file name).

**Comment by nul800sebastiaan** [comment]
@Aaen This is how Umbraco works, the media URL ends in the file name you uploaded. If you want to link to a media item in any of our built-in pickers it will store the media ID instead (https://github.com/umbraco/Umbraco-CMS/issues/4940) and from there in your template you can ask the media item for it's current file name. 

@ktzanoulinos 
> In the past and in the same version, the URL did NOT change. Suddenly and without any change code wise or version wise, the URL changes when we remove the file and upload a new one (with the exact same file name).

I can not reproduce this in version 8.9.1, there might be some custom code that is doing renames? Not sure.

**Comment by ktzanoulinos** [comment]
> I can not reproduce this in version 8.9.1, there might be some custom code that is doing renames? Not sure.

Hello, I am part of the agency that is implementing custom development on this project, so I can assure you that there is no custom development done that could result in this issue.
As I said, the customer informed us that suddenly the urls are changing when he is replacing the media file. Umbraco version is 8.5.5 and I can confirm since my team tested it.

**Comment by nul800sebastiaan** [comment]
@ktzanoulinos Sure! I just don't know what to tell you, this is a recording of me following the steps above in v8.5.5 and the URLs are exactly the same for me, what am I doing wrong?
![9481](https://user-images.githubusercontent.com/304656/101653818-0ae48b00-3a40-11eb-93d4-4ef0483eed8a.gif)

**Comment by dawoe** [comment]
@nul800sebastiaan does the url stay the same if you upload a different image ? And would be nice to hear from @ktzanoulinos which property editor they are using to select the media item in content.

**Comment by nul800sebastiaan** [comment]
As I said, when you upload a file with a different name, the URL will indeed change, that is expected behavior.

**Comment by dawoe** [comment]
Yep.. I know that. But in your gif it seems you upload the same image.

**Comment by ktzanoulinos** [comment]
We are using the native file media item provided by the CMS.
I have just made a test and I upload a different file, but with same file name and as you can see the URL is changing. This was not how it worked some weeks ago, and I suspect that Umbraco Cloud has changed something in its Azure storage that caused this issue. You can see the capture from the below video.
https://we.tl/t-wrsqjyrsHV

Thank you

**Comment by nul800sebastiaan** [comment]
Thanks again @ktzanoulinos - I wasn't aware that this was a Cloud site. I've talked to one of our developers there and it appears that this is a feature that was introduced to make sure that new uploads wouldn't get cached wrong. I'll re-open this and move it to the Cloud issue tracker. The team is probably going to disable the new blob storage provider for now until we can fix this specific issue.
